### ISSUE TITLE
Cannot get data from table after moving Realm Object Server to a different IP

### ISSUE THREAD

**Issue by LightningDev** [realm/realm-mobile-platform#83]
Hi,
I'm new to Realm Object Server so I don't know if I do something wrong or this is an issue. Everything works perfectly when I set up the server from scratch, from putting the data to Realm Server and run it from my iOS application (this is testing locally so the IP sometimes just change because my company reset the modem). After that I can still login using the Realm Browser, however, on iOS I cannot query any data from database anymore although I can still make a connection to Realm Object Server by using the admin account. Table always return 0 items.

The idea of my application is that it use the admin account to login to Realm Server and we have a table that stores all information of employee account including username and password, therefore, I need to retrieve and check it from that table to move to the next screen (just a formal checking credential method). 

I really appreciate if somebody tell me this is an issue or just my wrong logic. 

Thanks.

**Comment by Montrazul** [comment]
If the IP Address of your Realm Object Server changes you also have to change the IP Address inside your App Code. Did you do that?

**Comment by LightningDev** [comment]
Yes I did, that why I can authenticate the credential but I got nothing back from the database although the Realm Browser can access it.

**Comment by bigfish24** [comment]
Can you provide your client side code that is logging in a user and opening the synchronized Realm?

**Comment by istx25** [comment]
Hi @LightningDev! Would you be able to provide us with your client side code that is logging in a user and opening the synchronized Realm? Once we have that, we can better assist you.

**Comment by LightningDev** [comment]
@istx25 Hi, I'm on my holiday so I didn't check my email frequently. Here is what i'm doing in my application.
This is a login function:
`class func login(username: String, password: String, action: AuthenticationActions, completionHandler: @escaping ()->()) {
        let serverURL = NSURL(string: realmIP)!
        let credential = Credential.usernamePassword(username: username, password: password, actions: [action])
        if (syncUser == nil) {
            logoutAll()
            SyncUser.authenticate(with: credential, server: serverURL as URL) { user, error in
                if let user = user {
                    syncUser = user
                    let syncServerURL = URL(string: realmURL)!
                    setDefaultRealmConfigurationWithUser(user: user, serverURL: syncServerURL)
                } else if error != nil {
                    
                }
                completionHandler()
            }
        } else {
            completionHandler()
        }
    }
    class func setDefaultRealmConfigurationWithUser(user: SyncUser, serverURL: URL) {
        Realm.Configuration.defaultConfiguration = Realm.Configuration(
            syncConfiguration: (user, serverURL),
            objectTypes: [OPTOUser.self, Employee.self, Material.self, Contact.self, SalesOrder.self, PendingOrder.self, Group.self, Favourite.self])
        realm = try! Realm()
    }
`
After I login successfully using the syncUser, then I continue to retrieve the information of my created User table.
`    class func loginLocal(employee: String, password: String) -> Bool{
        let predicate = NSPredicate(format: "employee = %@ AND password = %@", employee, password)
        if (realm != nil) {
            let user = realm?.objects(OPTOUser.self).filter(predicate)
            if ((user?.count)! > 0) {
                return true
            }
        } 
        return false
    }`

**Comment by istx25** [comment]
No worries. Thank you for supplying us with this information. I will review what you've provided in the morning and see how we can proceed. Cheers! :)

**Comment by bigfish24** [comment]
@LightningDev I am sorry this slipped by on me... are you still having issues?

**Comment by LightningDev** [comment]
@bigfish24 
Hi,

I just found a solution and what cause my problems. I just added the download notification in the login progress to make sure that it will pull some data to the iPad so I can easily check the username and password inside my realm database. The Progress Notifications just make everything so simple and I also revamped some parts of my code to make sure it work in a same thread, so right now I don't have any issues much with my application.

Thank you so much.
